rust/main/chains/hyperlane-sealevel/src/log_meta_composer/tests.rs

```diff
@@ -49,7 +49,29 @@ pub fn test_search_dispatched_message_versioned_transaction() {
     );
 
     // then
-    assert!(!transaction_hashes.is_empty());
+    assert_eq!(transaction_hashes.len(), 1);
+}
+
+#[test]
+pub fn test_search_dispatched_message_relevant_instruction_out_of_two_in_single_transaction() {
+    // given
+    let mailbox_program_id = decode_pubkey("E588QtVUvresuXq2KoNEwAmoifCzYGpRBdHByN9KQMbi").unwrap();
+    let dispatched_message_pda_account =
+        decode_pubkey("HkS7U5adrqR4PZfn6DUEtHzwHxF8hZzmrEkJE8UuqFmz").unwrap();
+    let transactions = transactions(&read_json(
+        "dispatch_message_two_instructions_in_one_txn.json",
+    ));
+
+    // when
+    let transaction_hashes = search_transactions(
+        transactions,
+        &mailbox_program_id,
+        &dispatched_message_pda_account,
+        is_message_dispatch_instruction,
+    );
+
+    // then
+    assert_eq!(transaction_hashes.len(), 1);
```

Comment:
hm, maybe i'm misunderstanding here - is this always expected to be 1 because we specify the `dispatched_message_pda_account`? and the purpose of this unit test is to make sure we identify the second message? in that case I assume we want to make sure we find a tx hash for both pdas in the tx json

Reply:
There is a single transaction in the file used in this unit test. So, the possible outcomes are either 0 or 1 in terms of length of the vector we get back from the method.

Reply:
I see what you mean. The search should work for both messages. Let me check what should be done there.

Reply:
Added